src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java

```diff
@@ -1061,4 +1061,20 @@ public void testModuleNameInMultiThreadMode() throws Exception {
         }
     }
 
+    /**
+     * This test is a workaround for the Jacoco limitations.
+     * A call to {@link System#exit(int)} will never return, so Jacoco coverage probe will be
+     * missing. Mocking the {@code System} class will turn {@code System.exit()} to noop.
+     *
+     * @see <a href="https://github.com/jacoco/jacoco/issues/117">Jacoco issue 117</a>
+     * @throws Exception if error occurs
+     */
+    @Test
+    public void testJacocoWorkaround() throws Exception {
+        mockStatic(System.class);
+        Main.main();
+        verifyStatic(System.class, times(1));
+        System.exit(-1);
```

Comment:
I don't understand why `System.exit` is inside the test.

Reply:
So Powermock does validate calls to static methods.
A terrible syntax, but there is no alternative.

Reply:
@pbludov I am even more confused.
The point of the `System.exit` verify was to verify the call to that method in production code.
If we are verifying the call in test code, then it is completely pointless and should probably just be removed. The point of tests is to verify production code, not test code.

There is no way to verify `System.exit` is called in production?

Reply:
@rnveach yes it is confusing. The test verifies a call to `System.exit` in _production_ code.
To prove this, just comment out the line https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/Main.java#L241
The test will fail. Today i'll send a patch related to the Mockito2 API. You'll se a lot of such code.

Reply:
So why is this needed in test code then?

Reply:
@pbludov This has still not been resolved.

Reply:
done. The code reverted to the initial version.